pkg/geo/geo.go

```diff
@@ -617,6 +618,37 @@ func S2RegionsFromGeom(geomRepr geom.T, emptyBehavior EmptyBehavior) ([]s2.Regio
 // Common
 //
 
+// Normalize geographical coordinates
+// to spherical coordinates
+func makeValidGeographicalPoint(lat float64, lng float64) (float64, float64) {
+	if math.IsNaN(lat) || math.IsNaN(lng) {
+		return lat, lng
+	}
```

Comment:
can we add a check here to skip this logic if lat / lng is already in range?
we don't have to cast this into s2.LatLngFromDegrees btw.

```
func normalizeLngLat(lat float64, lng float64) (float64, float64) {
   if lat < -90 || lat > 90 {
      lat = NormalizeLatitudeDegrees(lat)
   }
   // same for lng
   return lat, lng
}
```

Reply:
Does your NormalizeLatitudeDegrees follow the logics found in in PostGIS? I was under impression that s2.LatLngFromDegrees does some preprocessing on the angles.
(https://github.com/postgis/postgis/blob/54399b9f6b0f02e8db9444f9f042b8d4ca6d4fa4/liblwgeom/lwgeodetic.c#L102-L155)

Reply:
The functions works as expected, updating the function as you suggested.